### PR TITLE
Add customer signinsignup on client side

### LINKED ISSUE #58: [Onboarding] Customer can signin/signup
Customers can sign in with their gpay accounts

### PR DESCRIPTION
**Pull request by hzxcaryn**
Closes #58 
To be merged after #88 and #96 

# Changes
- Customers can now login via the microapps identity API
- After calling the identity API, call our sever login endpoint to signin/signup the customer and get the customer details
- Added a temporary "Get Identity" button to trigger this login process
- Clicking that button will update the numCommits in the NumCommitsContext

# How to test
Visit the app via GPay Spot app. 
Click on the "Get Identity" button. Permissions dialog should be triggered and you should see the Commits badge appear.

Sample network request & response from inspect:
image

### CODE REVIEW

File: web/src/api.ts [hunk 1]
```diff
@@ -29,6 +29,31 @@ export const getCustomer = async (customerId: number): Promise<Customer> => {
   return res.json();
 };
 
+/**
+ * Logs in a customer using their
```
Comment (chshda) [inline web/src/api.ts:33]: Please complete the comment.

File: web/src/api.ts [hunk 2]
```diff
@@ -29,6 +29,31 @@ export const getCustomer = async (customerId: number): Promise<Customer> => {
...
+ * @param customerData
```
Comment (chshda) [inline web/src/api.ts:34]: Add more explanation to the parameter?

File: web/src/components/customer/explore/index.tsx [hunk 3]
```diff
@@ -46,6 +47,15 @@ const CustomerExplorePage: React.FC = () => {
     setNumCommits(numOngoingCommits);
   };
 
+  const handleGetIdentity = async () => {
+    const {
+      idToken,
+      decodedToken: {sub},
```
Comment (chshda) [inline web/src/components/customer/explore/index.tsx:53]: What does this "sub" mean here?

File: web/src/interfaces.ts [hunk 4]
```diff
@@ -57,11 +57,21 @@ export interface Listing {
...
 }
 
-export interface Customer {
-  id: number;
+/**
+ * CustomerPayload Interface that contains the fields of the payload that
+ * would be sent to the server.
+ */
+export interface CustomerPayload {
   gpayId: string; // E164 format
```
Comment (chshda) [inline web/src/interfaces.ts:65]: Why gpay ID has this format?

File: web/src/components/customer/explore/index.tsx (hunk 3 above)
Comment (hzxcaryn) [reply web/src/components/customer/explore/index.tsx:53]: @chshda `sub` is the unique identifier for gpay users! Can refer to https://developers.google.com/pay/spot/eap/reference/identity-api#identitytoken the `sub` field

File: web/src/api.ts (hunk 2 above)
Comment (hzxcaryn) [reply web/src/api.ts:34]: @chshda Omg 😅 Not sure how I missed it haha

File: web/src/interfaces.ts (hunk 4 above)
Comment (hzxcaryn) [reply web/src/interfaces.ts:65]: @chshda Parallax error 😆 Should be for the contactNumber

**Pull request merged**